kmip/core/attributes.py

```diff
@@ -354,6 +379,21 @@ def __validate(self):
                 msg += "; expected {0}, received {1}".format(
                     self.KeyRoleType, self.key_role_type)
                 raise TypeError(msg)
+        if self.digital_signature_algorithm is not None:
+            if not isinstance(self.digital_signature_algorithm,
+                              self.DigitalSignatureAlgorithm):
```

Comment:
You are correct in observing that this is done in other parts of the library. You are also correct that both versions are effectively the same; however, there is a difference. `self` refers directly to the instance of `CryptographicParameters` (which may vary from a vanilla instance depending on what's done with it by the instantiating code), while specifying `CryptographicParameters.` is a direct reference to the class definition (which is always what we want here). This actually ties into Python metaclass mechanics and how attributes are handled and accessed but that's a deep rabbit hole. To quote the Zen of Python, "Explicit is better than implicit."

I used to not really care about explicit vs implicit references. However, this ended up causing a nasty self reflection bug way back in an earlier version of the library (two years ago or so). Now, I try to avoid overusing `self`. One of my (many) TODOs is to go through and fix these references to explicitly use class names; I've just never gotten around to it. I also plan on pulling out these internal class definitions (e.g., `BlockCipherMode`, `PaddingMethod`, etc) and placing them in the outer module scope. Again, one of my many TODOs.